java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java

```diff
@@ -137,15 +138,35 @@ private Schema(@JsonProperty("fields") Iterable<Field> fields,
         .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
   }
 
+  static List<Map<String, String>> convertMetadata(Map<String, String> metadata) {
+    return (metadata == null) ? null : metadata.entrySet()
+        .stream()
+        .map(Schema::convertEntryToKeyValueMap)
+        .collect(Collectors.toList());
+  }
+
+  private static Map<String, String> convertEntryToKeyValueMap(Map.Entry<String, String> entry) {
+    Map<String, String> map = new HashMap<>(2);
+    map.put("key", entry.getKey());
```

Comment:
maybe we can extract the literals as static final variables?

Reply:
Done, I added constants for these keys and updated the unit tests to also reference those constants.